sample/app.py

```diff
@@ -93,55 +131,43 @@ def get_cached_data(key, fetch_func, expiry=3600):
+    cutoff_date = (end_date - timedelta(days=days))
+    
+    # Convert string dates to datetime objects
+    data_dict = {pd.to_datetime(k).date(): v for k, v in data_dict.items()}
+    
+    filtered_data = {k: v for k, v in data_dict.items() if k >= cutoff_date}
+    return [{'date': date.strftime('%Y-%m-%d'), **item} for date, item in sorted(filtered_data.items(), reverse=True)]
 
 @app.route('/stock/<symbol>')
 def stock_detail(symbol):
     tracer.start_as_current_span(inspect.currentframe().f_code.co_name)
-    end_date = datetime.now().date() - timedelta(days=1)  # Yesterday
-    start_date_1y = end_date - timedelta(days=365)
-    historical_data = fetch_historical_data(symbol, start_date_1y, end_date)
-    # Process the data for different time periods
-    def process_data(data_dict, days):
-        cutoff_date = (end_date - timedelta(days=days)).strftime('%Y-%m-%d')
-        filtered_data = {k: v for k, v in data_dict.items() if k >= cutoff_date}
-        sorted_data = sorted(filtered_data.items(), key=lambda x: x[0], reverse=True)
-        return [{
-            'date': date,
-            'open': item['open'],
-            'high': item['high'],
-            'low': item['low'],
-            'close': item['close'],
-            'volume': item['volume'],
-            'vwap': item['vwap']
-        } for date, item in sorted_data]
-
+    historical_data = get_stock_data(symbol)
     data = {
         '1M': process_data(historical_data, 30),
         '3M': process_data(historical_data, 90),
         '1Y': process_data(historical_data, 365)
     }
-    # Get latest available price with caching
-    def fetch_latest_price():
-        latest_trade = api.get_latest_trade(symbol)
-        return latest_trade.price
 
-    current_price = get_cached_data(f"{symbol}_latest_price", fetch_latest_price, expiry=300)  # Cache for 5 minutes
+    # Get latest available price
+    current_price = api.get_latest_trade(symbol).price
 
     # Calculate metrics
-    high_52week = max(item['high'] for item in historical_data.values())
-    low_52week = min(item['low'] for item in historical_data.values())
+    high_52week = max(item['High'] for item in historical_data.values())
+    low_52week = min(item['Low'] for item in historical_data.values())
     
-    # Calculate moving averages
-    closing_prices = [item['close'] for item in reversed(data['1Y'])]  # Reverse to get chronological order
-    ma_50 = sum(closing_prices[-50:]) / 50 if len(closing_prices) >= 50 else None
-    ma_200 = sum(closing_prices[-200:]) / 200 if len(closing_prices) >= 200 else None
+    # Calculate moving averages (already in the data)
+    last_data = list(historical_data.values())[-1]
+    ma_50 = last_data['SMA50']
+    ma_200 = last_data['SMA200']
     
     # Calculate percent change
-    prev_close = data['1Y'][1]['close']  # Second item is the previous day's close
+    prev_close = list(historical_data.values())[-2]['Close']
```

Comment:
What does this [-2] indexing refers ??
Can we find a way to make it more readable ??